Funcky/Extensions/EnumerableExtensions/Pairwise.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+
+namespace Funcky.Extensions
+{
+    public static partial class EnumerableExtensions
+    {
+        [Pure]
+        public static IEnumerable<(TSource Front, TSource Back)> Pairwise<TSource>(this IEnumerable<TSource> source)
+            => Pairwise(source, ValueTuple.Create);
+
+        [Pure]
+        public static IEnumerable<TResult> Pairwise<TSource, TResult>(this IEnumerable<TSource> source, Func<TSource, TSource, TResult> resultSelector)
+        {
+            using var enumerator = source.GetEnumerator();
+
+            if (enumerator.MoveNext())
+            {
+                for (var previous = enumerator.Current; enumerator.MoveNext(); previous = enumerator.Current)
```

Comment:
Couldn't the last part be omitted?
```csharp
for (var previous = enumerator.Current; enumerator.MoveNext();) // ...
```

Reply:
previous needs to be set to Current on each loop

Reply:
```csharp
for (var previous = enumerator.Current; enumerator.MoveNext(); previous = enumerator.Current)
//   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ but isn't it already set here?
```

Reply:
that is only executed once in the for loop

Reply:
Oooooh yes, how did I not see this?